src/common/botwrapper.ts

```diff
@@ -2,12 +2,13 @@ import { State } from 'boardgame.io';
 import { Bot } from 'boardgame.io/ai';
 import { BotAction } from 'boardgame.io/dist/types/src/ai/bot';
 
-type BotStrategy<T_SpecificGameState, T_Move> = (state: State<T_SpecificGameState>, botID: string) => T_Move|undefined;
+// Determine the next move for the bot and which move function to use.
+type BotStrategy<T_SpecificGameState, T_Move> = (state: State<T_SpecificGameState>, botID: string) => [T_Move|undefined, string];
```

Comment:
is the `T_` just a convention for custom types? I could not quickly find where the `T_Move` type is defined

Reply:
This is a generic type, like f(x)=x² defines the value x to define the result, the left-hand (`type BotStrategy<...> `) defines T_Move to define BotStrategy.

Similar to how `f(5)=25`, a concrete type is `BotStrategy<A,B>`, not BotStrategy itself.